week-1/Homework/mandatory/5-magic-8-ball.js

```diff
@@ -45,22 +45,61 @@ Very doubtful.
 
 // This should log "The ball has shaken!"
 // and return the answer.
-function shakeBall() {}
+function shakeBall() {
+  const shaken = "The ball has shaken!";
+  console.log(shaken);
+  return shaken;
+}
 
 // The answer should come from shaking the ball
-let answer;
+let answer = [
+  "It is certain.",
+  "It is decidedly so.",
+  "Without a doubt.",
+  "Yes - definitely.",
+  "You may rely on it.",
+  "As I see it, yes.",
+  "Most likely.",
+  "Outlook good.",
+  "Yes.",
+  "Signs point to yes.",
+  "Reply hazy, try again.",
+  "Ask again later.",
+  "Better not tell you now.",
+  "Cannot predict now.",
+  "Concentrate and ask again.",
+  "Don't count on it.",
+  "My reply is no.",
+  "My sources say no.",
+  "Outlook not so good.",
+  "Very doubtful.",
+];
+let randomAnswer = answer[Math.floor(Math.random() * answer.length)];
+console.log(randomAnswer);
 
 // When checking the answer, we should tell someone if the answer is
 // - very positive
 // - positive
 // - negative
 // - very negative
-function checkAnswer() {}
+
+function checkAnswer(answer) {
+  if (answer.indexOf(randomAnswer) <= 4) {
+    return "very positive";
+  } else if (answer.indexOf(randomAnswer) <= 9) {
+    return "positive";
+  } else if (answer.indexOf(randomAnswer) <= 14) {
+    return "negative";
+  } else {
+    return "very negative";
+  }
+}
+console.log(checkAnswer(answer));
 
 /* ======= TESTS - DO NOT MODIFY ===== */
 const log = console.log;
 let logged;
```

Comment:
You did well here. Some notes:
Remove the console logs after you finish testing your code. 
Do refactoring: In your case, you're passing an array that doesn't get used. 
 you can set the possible answers as an array in the beginning and make the shakeBall function the one that gets the random answer. Calling it will return the answer that you use in the next function. 
However, good work.

Reply:
Refactored as recommended. I have pushed all the changes to my local branch.
Thank you.

Reply:
Hmm, you really close, but you made some violations. We will discuss it tomorrow